README.md

```diff
@@ -896,3 +897,145 @@ def add_item_to_cart(cart, item):
 
 **[⬆ back to top](#table-of-contents)**
 
+# **Dependency Inversion Principle (DIP)**
+The DIP amounts to the following: 
+- *Depend upon abstractions. Do not depend upon concrete classes.*
+
+The object of this principle is to help decouple software and render high-level modules independant from low-level module implementation details. In detail, the principle states:
+1. High-level modules should not depend on low-level modules. Both should depend on abstractions (e.g. interfaces).
+2. Abstractions should not depend on details. Details (concrete implementations) should depend on abstractions.
+
+By dictating that both high-level and low-level objects must depend on the same abstraction, this design principle inverts the way some people may think about object-oriented programming.
+
```

Comment:
I think it would be nice to have another example to illustrate this point. Your example below for Mixins already does a good job of this though so up to you! I personally learn most from reading through examples with code and seeing how these theories and principles can be implemented so in my opinion, the more examples the better 😉

Reply:
Great point! I might add a visual for this :)